src/main/kotlin/io/github/mojira/arisa/ArisaMain.kt

```diff
@@ -57,33 +57,38 @@ fun main() {
     HelperMessageService.updateHelperMessages(helperMessagesFile)
     var helperMessagesLastFetch = Instant.now()
 
-    // Initialize caches and registry
-    val queryCache = Cache<List<Issue>>()
-    val moduleRegistry = ModuleRegistry(config)
-
-    val enabledModules = moduleRegistry.getEnabledModules().map { it.name }
-    log.debug("Enabled modules: $enabledModules")
-
-    // Create module executor
-    var moduleExecutor = ModuleExecutor(
-        config, moduleRegistry, queryCache,
-        getSearchIssues(jiraClient, config)
+    // Create executor
+    val moduleRegistries = getModuleRegistries(config)
+    var executor = Executor(
+        config, moduleRegistries, getSearchIssues(jiraClient, config)
     )
 
     while (true) {
-        // save time before run, so nothing happening during the run is missed
-        val curRunTime = Instant.now()
-        val executionResults = moduleExecutor.execute(lastRunTime, rerunTickets)
+        // Save time before run, so nothing happening during the run is missed
```

Comment:
Not for this PR, but a lot of this logic should be moved out of Main to its own service, Main should be as simple as possible and only orchestrating

Reply:
Yeah, it was already kinda tempting to put the `searchIssues` down below somewhere else, but I didn't want to do _too_ many intrusive changes here (especially since marcono's PR also exists).